### PR TITLE
Create Uphold Report

### LINKED ISSUE #2215: Track and Generate a report the KYC Date for a publisher
Business Development should be able to generate a report which contains the publisher_id, uphold_id, kyc_at information. 

edgecase: if they disconnect, the creation dates at our end stick so should not matter

### PR DESCRIPTION
**Pull request by corymcdonald**
## Create Uphold Report

For Business Development Purposes we will have a report that they can download via the admin dashboard to view all the uphold ids that were connected.

Because Uphold Connections can change owners and be disconnected it made more sense to create a separate db table that would track anytime a new connection was created. 

Closes #2215

### CODE REVIEW

File: db/migrate/20190910163930_create_uphold_report.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreateUpholdReport < ActiveRecord::Migration[5.2]
+  def change
+    add_column :uphold_connections, :member_at, :datetime
+
+    create_table :uphold_reports, id: :uuid, default: -> { "uuid_generate_v4()"}, force: :cascade do |t|
+      t.belongs_to :publisher, index: true, type: :uuid
+      t.uuid :uphold_id
```
Comment (yachtcaptain23) [inline db/migrate/20190910163930_create_uphold_report.rb]: Let's create an index here too to improve the above `find_by`

File: app/models/uphold_report.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class UpholdReport < ApplicationRecord
```
Comment (yachtcaptain23) [inline app/models/uphold_report.rb]: Nitpicking, but internally the convention that would be used is `UpholdStatusReport`

Comment (corymcdonald) [reply app/models/uphold_report.rb]: https://github.com/brave-intl/publishers/pull/2219/commits/c35cb3d2f276a0c1b81ee111a38d9f4b56e8744e

File: db/migrate/20190910163930_create_uphold_report.rb (hunk 1 above)
Comment (corymcdonald) [reply db/migrate/20190910163930_create_uphold_report.rb]: https://github.com/brave-intl/publishers/pull/2219/commits/c35cb3d2f276a0c1b81ee111a38d9f4b56e8744e

File: app/controllers/admin/uphold_reports_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        group('(EXTRACT(MONTH FROM created_at))::integer').
+        order('2 DESC, 3 DESC').count
+    end
+
+    def show
+      date = DateTime.strptime(params[:id], "%Y-%m")
+      start_date = date.at_beginning_of_month
+      end_date = date.at_end_of_month
+
+      @uphold_report = UpholdStatusReport.where("created_at >= :start AND created_at <= :finish", start: start_date, finish: end_date)
```
Comment (yachtcaptain23) [inline app/controllers/admin/uphold_reports_controller.rb]: This shouldn't be using `@`. Prefer renaming from `@uphold_report` to `uphold_status_report`.

File: app/controllers/admin/uphold_reports_controller.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+require 'csv'
+
+module Admin
+  class UpholdStatusReportsController < AdminController
+    def index
+      @uphold_report = UpholdStatusReport.
```
Comment (yachtcaptain23) [inline app/controllers/admin/uphold_reports_controller.rb]: Prefer renaming to `@uphold_status_reports`

File: app/views/admin/uphold_reports/index.html.slim [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  p Anytime a publisher connects their uphold account we create an entry in our database that logs this. If an uphold id has already been reported we do not create an entry. This data is then aggregated and grouped by month and displayed here.
+
+table.table
+  thead
+    tr
+      td Period
+      td Number of users
+      td
+  tbody
+    - @uphold_report.each do |report|
```
Comment (yachtcaptain23) [inline app/views/admin/uphold_reports/index.html.slim]:
```suggestion
    - @uphold_reports.each do |report|
```

File: app/controllers/admin/uphold_reports_controller.rb (hunk 3 above)
Comment (yachtcaptain23) [inline app/controllers/admin/uphold_reports_controller.rb]:
```suggestion
      uphold_status_report = UpholdStatusReport.where("created_at >= :start AND created_at <= :finish", start: start_date, finish: end_date)
```

File: app/controllers/admin/uphold_reports_controller.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      date = DateTime.strptime(params[:id], "%Y-%m")
+      start_date = date.at_beginning_of_month
+      end_date = date.at_end_of_month
+
+      @uphold_report = UpholdStatusReport.where("created_at >= :start AND created_at <= :finish", start: start_date, finish: end_date)
+
+      generated = []
+      generated << ["publisher id", "publisher created at", "uphold id", "uphold connected time"].to_csv
+
+      @uphold_report.each do |report|
```
Comment (yachtcaptain23) [inline app/controllers/admin/uphold_reports_controller.rb]:
```suggestion
      uphold_status_reports.each do |report|
```

File: app/controllers/admin/uphold_reports_controller.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
+require 'csv'
+
+module Admin
+  class UpholdStatusReportsController < AdminController
+    def index
+      @uphold_status_reports = UpholdStatusReport.
+        group('(EXTRACT(YEAR FROM created_at))::integer').
+        group('(EXTRACT(MONTH FROM created_at))::integer').
+        order('2 DESC, 3 DESC').count
```
Comment (yachtcaptain23) [inline app/controllers/admin/uphold_reports_controller.rb]:
```suggestion
        order('2 DESC, 3 DESC')
```

Comment (corymcdonald) [reply app/controllers/admin/uphold_reports_controller.rb]: Why this change? Since we're doing a `group` I want get a count of the amount in each report

**Pull request merged**